### PR TITLE
Add post about contributing to the GOV.UK Design System

### PR DESCRIPTION
**Pull request by frankieroberto**
Thought I’d write this up whilst it was fresh in my mind. Feedback welcome! Is this helpful to others thinking about contributing?

Thoughts and ideas for the title especially welcome!

### CODE REVIEW

File: posts/contributing-to-govuk-design-system.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+title: Why I contribute to the GOV.UK Design System (and why you should too)
+date: 2022-05-03
+tags: post
+layout: post
+author:
+  name: Frankie Roberto
+  url: https://github.com/frankieroberto
+---
+
+Last week a new pattern was published in the GOV.UK Design System that helps users to [confirm a phone number](https://design-system.service.gov.uk/patterns/confirm-a-phone-number/) using a security code sent by text message.
```
Comment (paulrobertlloyd) [inline posts/contributing-to-govuk-design-system.md:11]:
```suggestion
Last week a new pattern was published in the GOV.UK Design System that helps users [confirm a phone number](https://design-system.service.gov.uk/patterns/confirm-a-phone-number/) using a security code sent by text message.
```

File: posts/contributing-to-govuk-design-system.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+I initiated this contribution almost two years ago when working on a service that was switching to using its own sign in system to avoid relying upon an off-the-shelf authentication service which was hard to maintain.
+
+As the service was quite sensitive, we decided to require a text message confirmation code when signing in – sometimes known as 'two factor authentication'.
```
Comment (paulrobertlloyd) [inline posts/contributing-to-govuk-design-system.md:15]: Don’t think you need scare quotes:

```suggestion
As the service was quite sensitive, we decided to require a text message confirmation code when signing in – sometimes known as two factor authentication.
```

File: posts/contributing-to-govuk-design-system.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+The pattern of sending and requesting an SMS code is quite well established, used by many services in and out of government.
+
+However we still had a bunch of detailed design questions. Things like:
+
+* should we call it a ‘security code’, a ‘verification code’, or something else?
```
Comment (paulrobertlloyd) [inline posts/contributing-to-govuk-design-system.md:21]: A content designer friend always told me these were unnecessary:

```suggestion
* should we call it a security code, a verification code, or something else?
```

File: posts/contributing-to-govuk-design-system.md [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+* how long should the codes be valid for?
+* if a code doesn’t arrive straight away and the user requests a new one, should we resend the same code or generate a new one?
```
Comment (paulrobertlloyd) [inline posts/contributing-to-govuk-design-system.md:23]: Truly nitpicky details, but I’ve been sticking to dump quotes and letting Smartypants to the hard work (on this website, anyway): 

```suggestion
* if a code doesn't arrive straight away and the user requests a new one, should we resend the same code or generate a new one?
```

File: posts/contributing-to-govuk-design-system.md [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+* when first setting up an account, how we can we make sure the phone number is valid and can receive text messages without letting the user get stuck?
+
+For each of these questions, we looked at what existing services like GOV.UK Notify and HMRC did, and spoke to some of their designers to try to understand the reasons behind the decision they’d taken. Being able to reach out to the wider government design community like this, through platforms like the cross-government Slack and Twitter, is super valuable.
+
+There was also an existing thread on the GOV.UK Design System backlog for [identifying users](https://github.com/alphagov/govuk-design-system-backlog/issues/25), where designers and researchers had shared some screenshots and discussed the pattern. These [backlog threads](https://github.com/alphagov/govuk-design-system-backlog/issues) are another great resource, if sometimes hard to find.
+
+Once we had assimilated some of this existing work, plus made some design iterations of our own, we were ready to ship our feature. And once the dust had settled, I realised that writing this all up might be useful for other designers in the same situation.
```
Comment (paulrobertlloyd) [inline posts/contributing-to-govuk-design-system.md:31]: Double ‘once’ in this sentence is tripping me up, but not sure of a good alternative. 🤔

**Review by paulrobertlloyd — COMMENTED** [review]
Love it! Good advice, and perhaps helps to set the tone for the website and future posts!

File: posts/contributing-to-govuk-design-system.md [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
```
Comment (frankieroberto) [inline posts/contributing-to-govuk-design-system.md:31]:
```suggestion
When we had assimilated some of this existing work, plus made some design iterations of our own, we were ready to ship. And once the dust had settled, I realised that writing this all up might be useful for other designers in the same situation.
```

Comment (frankieroberto) [reply posts/contributing-to-govuk-design-system.md:31]: @paulrobertlloyd  how’s this, switching the first one to ‘when’ (as ‘once the dust had settled’ is idiomatic).

Also realised that “our feature” doesn’t add anything.

File: posts/contributing-to-govuk-design-system.md (hunk 5 above)
Comment (frankieroberto) [reply posts/contributing-to-govuk-design-system.md:31]: Good point! See suggestion below (or above, depending on GitHub UI).

File: posts/contributing-to-govuk-design-system.md (hunk 6 above)
Comment (paulrobertlloyd) [reply posts/contributing-to-govuk-design-system.md:31]: 👌

File: posts/contributing-to-govuk-design-system.md [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+author:
+  name: Frankie Roberto
+  url: https://github.com/frankieroberto
+---
+
+Last week a new pattern was published in the GOV.UK Design System that helps users [confirm a phone number](https://design-system.service.gov.uk/patterns/confirm-a-phone-number/) using a security code sent by text message.
+
+I initiated this contribution almost two years ago when working on a service that was switching to using its own sign in system to avoid relying upon an off-the-shelf authentication service which was hard to maintain.
+
+As the service was quite sensitive, we decided to require a text message confirmation code when signing in – sometimes known as two factor authentication.
```
Comment (fofr) [inline posts/contributing-to-govuk-design-system.md:15]:
```suggestion
As the service was sensitive, we chose to use a text message confirmation code when signing in – sometimes known as two factor authentication.
```

**Review by fofr — COMMENTED** [review]
Nice post @frankieroberto, one minor content suggestion from me

File: posts/contributing-to-govuk-design-system.md [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+
+The pattern of sending and requesting an SMS code is quite well established, used by many services in and out of government.
+
+However we still had a bunch of detailed design questions. Things like:
+
+* should we call it a security code, a verification code, or something else?
+* how long should the codes be valid for?
+* if a code doesn't arrive straight away and the user requests a new one, should we resend the same code or generate a new one?
```
Comment (ralph-hawkins) [inline posts/contributing-to-govuk-design-system.md:23]:
```suggestion
* if a code doesn’t arrive straight away and the user requests a new one, should we resend the same code or generate a new one?
```

Comment (frankieroberto) [reply posts/contributing-to-govuk-design-system.md:23]: There's some code called 'Smartypants' which auto-converts these to typographic apostrophes, I think. See https://github.com/x-govuk/x-govuk.github.io/pull/15#discussion_r841280829

**Pull request merged**